src/flutter/shell/platform/linux_embedded/surface/environment_egl.h

```diff
@@ -28,8 +28,8 @@ class EnvironmentEgl {
 
   EnvironmentEgl() : display_(EGL_NO_DISPLAY), valid_(false) {}
 
-  ~EnvironmentEgl() {
-    if (display_ != EGL_NO_DISPLAY) {
+  virtual ~EnvironmentEgl() {
```

Comment:
If all destructors in the derived class are 'default' and there are no other virtual functions, I don't think you need to make this a virtual destructor.

Reply:
Deleted virtual.